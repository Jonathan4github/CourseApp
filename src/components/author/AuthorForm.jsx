import React from 'react';
import PropTypes from 'prop-types';

import TextInput from '../common/TextInput';


const propTypes = {
  author: PropTypes.shape().isRequired,
  handleOnSave: PropTypes.func.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  handleOnFocus: PropTypes.func.isRequired,
  isSaving: PropTypes.bool.isRequired,
  errors: PropTypes.shape()
};

const defaultProps = {
  errors: {}
};

const AuthorForm = ({
  author, handleOnSave, handleOnChange, handleOnFocus, isSaving, errors
}) => (
  <form>
    <TextInput
      name="firstName"
      label="First Name"
      value={author.firstName}
      onChange={handleOnChange}
      onFocus={handleOnFocus}
      placeholder="eg. John"
      error={errors.firstName} />
    <TextInput
      name="lastName"
      label="Last Name"
      value={author.lastName}
      onChange={handleOnChange}
      onFocus={handleOnFocus}
      placeholder="eg. Doe"
      error={errors.lastName} />
    <input
     type="submit"
     className="btn btn-primary"
     disabled={isSaving}
     value={isSaving ? 'Saving...' : 'Save'}
     onClick={handleOnSave} />
  </form>
);
AuthorForm.propTypes = propTypes;
AuthorForm.defaultProps = defaultProps;

export default AuthorForm;
