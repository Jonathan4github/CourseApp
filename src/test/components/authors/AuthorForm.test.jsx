/* eslint-disable */
import AuthorForm from '../../../components/author/AuthorForm';
import authorMock from '../../__mocks__/authorMock';


describe('<AuthorForm />', () => {
  const setup = () => {
    const props = {
      author: authorMock.author,
      handleOnSave: jest.fn(),
      handleOnChange: jest.fn(),
      handleOnFocus: jest.fn(),
      isSaving: false,
      errors: {}
    };

    return {
      shallowWrapper: shallow(<AuthorForm {...props} />),
      props
    };
  };

  it('renders component without crashing', () => {
    const { shallowWrapper } = setup();

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should set button input text to "Saving..."', () => {
    const { shallowWrapper, props } = setup();
    shallowWrapper.setProps({
      ...props,
      isSaving: true
    });

    expect(shallowWrapper.find('input').props().value).toEqual('Saving...');
  });

  it('renders two(2) <TextInput /> child components', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper.find('TextInput').length).toBe(2);
  });

  it('calls handleOnSave method when form is submitted', () => {
    const { shallowWrapper, props } = setup();

    shallowWrapper.find('input').simulate('click');
    expect(props.handleOnSave).toHaveBeenCalled();
  });
});
