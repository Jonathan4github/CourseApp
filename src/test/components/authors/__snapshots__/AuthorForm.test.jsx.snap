// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorForm /> renders component without crashing 1`] = `
<form>
  <TextInput
    error=""
    label="First Name"
    name="firstName"
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="eg. John"
    value="Cory"
  />
  <TextInput
    error=""
    label="Last Name"
    name="lastName"
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="eg. Doe"
    value="House"
  />
  <input
    className="btn btn-primary"
    disabled={false}
    onClick={[MockFunction]}
    type="submit"
    value="Save"
  />
</form>
`;
